### PR TITLE
[SR GeekCamp] Support manual refresh hive external table schema info

### PR DESCRIPTION
**Pull request by Zhangruichao**
## What type of PR is this：
- [ ] feature

## Which issues of this PR fixes ：

related with  #4030

## Problem Summary(Required) ：
 1. Add code for judging hive schema change
 2. Refresh table schema cache
 3. Persist the schema change log

### CODE REVIEW

File: fe/fe-core/src/main/java/com/starrocks/catalog/HiveTable.java [hunk 1]
```diff
@@ -191,11 +194,80 @@ public Map<String, HiveColumnStats> getTableLevelColumnStats(List<String> column
...
+        if (!needRefreshColumn) {
+            refreshTableCache(nameToColumn);
+        } else {
+            refreshSchemaCache(preNameToColumn, updatedTableSchemas);
+            refreshTableCache(nameToColumn);
+            modifyTableSchema(dbName, tableName);
+        }
+    }
+
+    private void refreshTableCache(Map<String, Column> nameToColumn) throws DdlException {
```
Comment (stephen-shelby) [inline fe/fe-core/src/main/java/com/starrocks/catalog/HiveTable.java:210]: LGTM

**Comment by stephen-shelby** [comment]
old pr refer to : https://github.com/StarRocks/starrocks/pull/4319

**Review by gengjun-git — DISMISSED** [review]

**Comment by gengjun-git** [comment]
run starrocks_fe_unittest

**Comment by gengjun-git** [comment]
run starrocks_fe_unittest

**Comment by gengjun-git** [comment]
run starrocks_fe_unittest

**Review by gengjun-git — APPROVED** [review]

**Review by stephen-shelby — APPROVED** [review]

**Review by caneGuy — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 14 / 105 (13.33%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/persist/ModifyTableColumnOperationLog.java|0|12|00.00%|
|:large_blue_circle:|com/starrocks/catalog/HiveMetaStoreTable.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/server/GlobalStateMgr.java|0|14|00.00%|
|:large_blue_circle:|com/starrocks/persist/EditLog.java|0|6|00.00%|
|:large_blue_circle:|com/starrocks/journal/JournalEntity.java|0|3|00.00%|
|:large_blue_circle:|com/starrocks/catalog/HiveTable.java|14|69|20.29%|

**Pull request merged**